src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,273 @@
+import { FiatAccountSchema, FiatConnectError } from '@fiatconnect/fiatconnect-types'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch } from 'react-redux'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import BorderlessButton from 'src/components/BorderlessButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import TextInput from 'src/components/TextInput'
+import { addNewFiatAccount } from 'src/fiatconnect'
+import i18n from 'src/i18n'
+import ForwardChevron from 'src/icons/ForwardChevron'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'FiatDetailsScreen'
+type ScreenProps = StackScreenProps<StackParamList, Screens.FiatDetailsScreen>
+
+type Props = ScreenProps
+
+interface Fields {
+  name: string
+  label: string
+  regex: RegExp
+  placeholderText: string
+  errorMessage: string
+}
+
+// This is a mapping between different fiat account schema to the metadata of the fields that need to be rendered on the bank details screen
+const SCHEMA_TO_FIELD_METADATA_MAP = {
+  AccountNumber: [
```

Comment:
to ensure it matches
```suggestion
  [FiatAccountSchema.AccountNumber]: [
```